### ISSUE TITLE
Cleanup unnecessary nested promises in services.

### ISSUE THREAD

**Issue by tracker1** [johnbwoodruff/digitalocean-js#57]
I created a fork of this repo, in order to use it with Deno at [tracker1/digitalocean-deno](https://github.com/tracker1/digitalocean-deno).

One of the things I did, is replace axios with fetch, the other is I cleaned up the services in order to remove the unnecessary nested promises.  If you were interested adapting back for use in this repo should be relatively easy... mostly, copying the lib folder, and from there replacing the `include... from '*.ts'` removing the `.ts` file suffix.  I also renamed the polyfill I made with fetch to `request`, so I had to rename some of the `request` data variables to `data`.

If you're open to merging these changes back in, I can create a pull request for you... the changes to fetch aren't strictly necessary, and could keep axios (changing the instance export name, and keeping the setup), alternatively could use the `node-fetch` module, and remove it entirely for browsers, and for node after 18.x becomes the LTS release, if you wanted to.

I'm hoping to get it configured for a publish to Denoland ... and possibly getting a browser and node build path... I notice there's only one test file in place, so will try and port those to `deno test` syntax.  If you're at all interested in working together, I'm happy to do so... It seems you know the DO API far better than I do... I mostly just cleaned, removed some extra bits, and minor cleanup.

**Comment by tracker1** [comment]
@johnbwoodruff understandable... also have a todo list a mile long for myself on top of the day job.  Will keep an eye out to more replies in this thread.

For now, I was able to get the library working for my immediate needs, which was the main thing at least for my own use case.  Kind of wild how much the JS/TS ecosystem has shifted over time...  It's almost another dark ages trying to keep what goes into the browser, deno, node etc all working in something resembling reusable code.  I'm really liking Deno a lot more than Node in so many ways, but the rich npm packages is something that's not the easiest thing to get over, and some of the bridge options aren't great for modern(ish) packages.

**Comment by tracker1** [comment]
@johnbwoodruff ping

**Comment by johnbwoodruff** [comment]
Looping back on this, I finally completed a fairly major overhaul to the project structure to convert it to an NX workspace instead of the custom typescript seed project I originally used. I've also begun the process of writing some tests for the library. As part of that project overhaul, since I was changing everything anyway, I also went through and unwrapped the promises like you requested. Great suggestion, thanks! Closed in #49.
